File: C#/teste-prato/teste-prato/Utils/FilaPerguntas.cs
using System.Collections.Generic;
using teste_prato.Models;

namespace teste_prato.Utils
{
    public class FilaPerguntas
    {
        private List<Pergunta> elements { get; set; }
        private int count = 0;


        public int getCount()
        {
            return count;
        }


        public void setCount(int count)
        {
            this.count = count;
        }


        public FilaPerguntas()
        {
            elements = new List<Pergunta>();

        }


        public bool push(Pergunta novaPergunta)
        {
            int indice = elements.Count == 0 ? 0 : elements.Count - 1;
            elements.Insert(indice, novaPergunta);
            return true;
        }


        public Pergunta top()
        {
            if (elements.Count == 0) return null;

            Pergunta pergunta = elements[0];

            return pergunta;
        }

        public Pergunta get(int index)
        {
            if (elements.Count == 0) return null;

            Pergunta pergunta = elements[index];

            return pergunta;
        }

        public Pergunta next()
        {
            if (elements.Count == 0) return null;

            Pergunta pergunta = elements[count];
            count++;
            return pergunta;
        }

        public Pergunta tail()
        {
            if (elements.Count == 0) return null;

            Pergunta pergunta = elements[elements.Count - 1];
            return pergunta;
        }

        public int size()
        {
            return elements.Count;
        }
    }
}

File: Java/teste-prato/src/br/com/pratoTest/utils/FilaPerguntas.java
package br.com.pratoTest.utils;

import java.util.ArrayList;

import br.com.pratoTest.Model.Pergunta;
import br.com.pratoTest.Model.Prato;

public class FilaPerguntas {
	
	private ArrayList<Pergunta> elements; 
	private int count = 0;


	public int getCount() {
		return count;
	}


	public void setCount(int count) {
		this.count = count;
	}


	public FilaPerguntas() {
		elements = new ArrayList<Pergunta>(); 
		
	}


	public boolean push(Pergunta novaPergunta) {
		
         elements.add(elements.size() == 0 ? 0 : elements.size()-1, novaPergunta); 
         return true;
	}


	public Pergunta top() {
		if (elements.size() == 0) return null;
		
		Pergunta pergunta = elements.get(0);
		
		return pergunta;
	}
	
	public Pergunta get(int index) {
		if (elements.size() == 0) return null;
		
		Pergunta pergunta = elements.get(index);
		
		return pergunta;
	}
	
	public Pergunta next() {
		if (elements.size() == 0) return null;
		
		Pergunta pergunta = elements.get(count);
		count++;
		return pergunta;
	}
	
	public Pergunta tail() {
		if (elements.size() == 0) return null;
		
		Pergunta pergunta = elements.get(elements.size()-1);
		return pergunta;
	}

	public int size() {
		return elements.size();
	}

}
